src/mote/templates/mainpage.html

```diff
@@ -377,8 +385,129 @@ <h6 class="modal-title head fw-bold">Recent conversations</h6>
+            dropdownDiv.classList.add("show");
+            search_results();
+        });
+        async function populate_channels(data) {
+            data = JSON.parse(decodeURIComponent(data))
+            initialize_search_modal();
+            document.getElementsByClassName("input-group mb-3")[0].innerHTML = ""
+            document.getElementById("findtext").value = "";
+            document.getElementById("none-find").innerHTML = "";
+            document.getElementById("find-cont").innerHTML = `
+                    <div class="text-center text-muted mt-1 mb-1">${data.length} conversation(s) found</div>
+                    <ul class="list-group" id="listfind-uols"></ul>
+                `;
+            document.getElementById("find-init").innerHTML = "";
+            for (let indx in data) {
+                $("#listfind-uols").append(`
+                        <a class="list-group-item list-group-item-action" 
+                        type="button" 
+                        href="${data[indx]["slug"]["summary"]}" 
+                        target="_blank">
+                            <div class="head h4 ellipsis">
+                                <span>${data[indx]["topic"]}</span>
+                            </div>
+                            <div class="fst-italic small text-muted mt-1">
+                                <i class="fas fa-clock"></i>&nbsp;${data[indx]["time"]}&nbsp;
+                                <i class="fas fa-calendar"></i>&nbsp;${data[indx]["date"]}&nbsp;
+                                <i class="fas fa-layer-group"></i>&nbsp;${data[indx]["channel"]}&nbsp;
+                            </div>
+                        </a>
+                    `);
+            }
+        }
+
+        function debounce(func, timeout = 300) {
+            let timer;
+            return (...args) => {
+                clearTimeout(timer);
+                timer = setTimeout(() => {
+                    func.apply(this, args);
+                }, timeout);
+            };
+        }
+
+        const search_results = debounce(() => fetch_search_results());
+
+        let prevValue = ""
+        async function fetch_search_results() {
+            let value = document.getElementById("findtext").value.toLowerCase();
+            if (value != "") {
+                if (value.length > 3 && value !== prevValue) {
+                    var dropdownDiv = document.getElementById('dropdown');
+                    dropdownDiv.innerHTML =
+                        `<li><a class="dropdown-item" type="button">Loading...</a></li>`
+                    dropdownDiv.classList.add("show")
+                    prevValue = value
+                    await $.getJSON("/fragedpt/", {
+                        "rqstdata": "srchmeet",
+                        "srchtext": value
+                    }, function (data) {
+                        results = data
+                        console.log(data);
```

Comment:
I assume you used that for debugging purpose. If that is the case, you can remove it if it's not needed anymore.

Reply:
sure, yes I used it for testing purpose